src/backend/tests/kernelprog/vecadd/vecadd.driver.sexp

```diff
@@ -0,0 +1,146 @@
+(c-lisp
+    ;; External linkage
+    (define ((getchar int8)))
+    (define ((exit void) (status int)))
+    (define ((malloc (ptr float)) (sz int)))
+    (define ((rand float)))
+    ; From <math.h> (-lm)
+    (define ((fabsf float) (a float)))
+    (define ((fmaxf float) (a float) (b float)))
+    (define ((free void) (p (ptr float))))
+
+    ; Wrapper around `printf` from <stdio.h>
+    ,(define_printf)
+
+    ; Function signatures and type aliases from Numba's CUDA driver
+    ,@(get_cuda_signatures)
+
+    (define ((ref_kernel void) (a (ptr float)) (b (ptr float)) (res (ptr float)) (N int))
+        (declare i int)
+        (for ((set i 0)
+              (lt i N)
+              (set i (add i 1)))
+            (store (ptradd res i)
+                   (fadd (load (ptradd a i))
+                         (load (ptradd b i))))))
+
+    (define ((error_check void) (res int) (call_str (ptr int8)))
+        (if (ne res 0)
+            (,(printf call_str)
```

Comment:
haha printf is now a proper macro!

Reply:
Yes! We don't even need our old `print` utility now.

Reply:
yayyy